src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -235,7 +257,7 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                         {
                             transitiveOrigins.TryGetValue(packageRef.PackageIdentity.Id, out TransitiveEntry cacheEntry);
                             return MergeTransitiveOrigin(packageRef, cacheEntry);
-                        });
+                        }).ToList();
```

Comment:
When `transitivePackagesWithOrigins` is used a few lines down, `.ToList()` is called there. So doing so here causes a second List to be allocated (and depending on the implementation of LINQ's ToList, if it doesn't preallocate a list of the correct size, then it will cause multiple resizes, which will be very costly).